trie/trie.go

```diff
@@ -0,0 +1,57 @@
+package trie
```

Comment:
We are missing some proper documentation on how to use the `trie` package. Mentioning that it isn't concurrent safe should also be done. You don't need anything particular fancy, look at the cachettl package for an example: https://github.com/rudderlabs/rudder-go-kit/blob/v0.48.2/cachettl/cachettl.go#L8